analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantConditionalAroundAssignment.cs

```diff
@@ -36,58 +38,80 @@ public sealed class RedundantConditionalAroundAssignment : SonarDiagnosticAnalyz
-
-                    var expression1Condition = condition.Left?.RemoveParentheses();
-                    var expression2Condition = condition.Right?.RemoveParentheses();
-                    var expression1Assignment = assignment.Left?.RemoveParentheses();
-                    var expression2Assignment = assignment.Right?.RemoveParentheses();
-
-                    if (!AreMatchingExpressions(expression1Condition, expression2Condition, expression2Assignment, expression1Assignment) &&
-                        !AreMatchingExpressions(expression1Condition, expression2Condition, expression1Assignment, expression2Assignment))
-                    {
-                        return;
-                    }
-
-                    if (!(c.SemanticModel.GetSymbolInfo(assignment.Left).Symbol is IPropertySymbol))
-                    {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, condition.GetLocation()));
-                    }
-                },
-                SyntaxKind.IfStatement);
+            context.RegisterSyntaxNodeActionInNonGenerated(UselessConditionIfStatement, SyntaxKind.IfStatement);
+            context.RegisterSyntaxNodeActionInNonGenerated(UselessConditionSwitchExpression, SyntaxKindEx.SwitchExpression);
+        }
+
+        private static void UselessConditionIfStatement(SyntaxNodeAnalysisContext c)
+        {
+            var ifStatement = (IfStatementSyntax)c.Node;
+
+            if (ifStatement.Else != null
+                || ifStatement.Parent is ElseClauseSyntax
+                || (ifStatement.FirstAncestorOrSelf<AccessorDeclarationSyntax>()?.IsKind(SyntaxKind.SetAccessorDeclaration) ?? false)
+                || !TryGetNotEqualsCondition(ifStatement, out var condition)
+                || !TryGetSingleAssignment(ifStatement, out var assignment))
+            {
+                return;
+            }
+
+            var expression1Condition = condition.Left?.RemoveParentheses();
+            var expression2Condition = condition.Right?.RemoveParentheses();
+            var expression1Assignment = assignment.Left?.RemoveParentheses();
+            var expression2Assignment = assignment.Right?.RemoveParentheses();
+
+            if (!AreMatchingExpressions(expression1Condition, expression2Condition, expression2Assignment, expression1Assignment)
+                && !AreMatchingExpressions(expression1Condition, expression2Condition, expression1Assignment, expression2Assignment))
+            {
+                return;
+            }
+
+            if (!(c.SemanticModel.GetSymbolInfo(assignment.Left).Symbol is IPropertySymbol))
+            {
+                c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, condition.GetLocation()));
+            }
+        }
+
+        private static void UselessConditionSwitchExpression(SyntaxNodeAnalysisContext c)
+        {
+            var switchExpression = (SwitchExpressionSyntaxWrapper)c.Node;
+
+            if (switchExpression.Arms.Count != 1
```

Comment:
Because of this check we do not cover something like:
```
int y = 6;

y = y switch
{
    5 => 5, // Noncompliant
    6 => 6 // Noncompliant
};
```